tests/unit/test_xblock.py

```diff
@@ -0,0 +1,70 @@
+import sys
+
+if sys.version_info[0] >= 3:
+    import unittest
+    from unittest import mock as mock
+else:
+    import mock as mock
+    import unittest2 as unittest
+
+from mock import patch, MagicMock
+
+from requests.exceptions import Timeout
+
+from vimeo.vimeo import VimeoBlock
+
+from tests import createVimeoXBlockTestInstance
+
+class TestXBlock(unittest.TestCase):
+    """
+    Unit tests for `vimeo_xblock`
+    """
+    def setUp(self):
+        self.vimeo_block = createVimeoXBlockTestInstance()
+
+    @patch("requests.get")
+    def test_vimeo_url_get_embeded_code(self, mock_request):
+        mock_request.return_value.status_code = 200
+
+        url = "https://vimeo.com/306558851"
+        _, res = self.vimeo_block.get_embed_code_for_url(url)
+
+        self.assertNotIn(self.vimeo_block.exception_error_msg, res)
+        self.assertNotIn(self.vimeo_block.unsupported_error_msg, res)
+
+    @patch("requests.get")
+    def test_youtube_url_get_embedded_code(self, mock_request):
+        mock_request.return_value.status_code = 200
+
+        url = "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
+        _, res = self.vimeo_block.get_embed_code_for_url(url)
+
+        self.assertIn(self.vimeo_block.unsupported_error_msg, res)
+
+    @patch("requests.get")
+    def test_bad_url_get_embedded_code(self, mock_request):
+        mock_request.side_effect = Timeout()
+
+        url = "https://vimeo.com/306558851"
+        _, res = self.vimeo_block.get_embed_code_for_url(url)
+
+        self.assertIn(self.vimeo_block.exception_error_msg, res)
+
+    @patch("{0}.VimeoBlock.create_fragment".format(__name__))
+    @patch("{0}.VimeoBlock.get_embed_code_for_url".format(__name__))
+    def test_student_view(self, mock_embeded_code, mock_create_fragment):
+        mock_embeded_code.return_value = ("", "")
+        mock_create_fragment.return_value = True
+
+        self.assertTrue(self.vimeo_block.student_view(None))
```

Comment:
Please add a proper return_value to mocked method and verify them with the value returned in student_view so it's more clear what is being tested.

Reply:
I'll add an assert that it was called once instead of adding a return value. Since it should only be called once. What do you think @pkulkark?

Reply:
The `mock_embeded_code` is already being verified before. Therefore, there is no need to verify it because it already has unit tests. The `create_fragment` requires Django in order to render the things. Thus, I think creating an assert called once would be a good solution since it is called with a return statement.

Reply:
It's fine to add a return value since you're mocking the methods. The main idea is to verify that the student_view returns the correct value.

Reply:
I can't verify that student_view is returning the correct value because it requires that I render the template using `create_fragment`. And I can't test `create_fragment` without creating a runtime (the integration test). I already have an integration test setup for `create_fragment`. I just need to setup a runtime for it (but this will result in going overboard than the estimated time by like an hour).

But it is important to note that the `create_fragment` is a function which wouldn't be changed, it is kind of by the book. I am aware that it needs to be tested, however. But I can't test it without an instance.

> Actually I just had an idea, which might take 15 mins max. I will try to apply it. I'll mock django's rendering. And I'll verify that the fragment has the HTML code. I am not sure 100% it will work, but I'll give it a try.

Reply:
You don't need to test create_fragment here. The idea is to give a sense of what value we would see if this ran on an actual runtime. So instead of `self.assertTrue(self.vimeo_block.student_view(None))` you will assert it with what an actual result looks like (which is added to the return_value of the mocked create_fragment method). Basically it's for visualization. I hope that's clear?

Reply:
Yes, that's actually really clear. I'll work on it, thank you!